insights/parsers/spamassassin_channels.py

```diff
@@ -0,0 +1,44 @@
+"""
+SpamassassinChannels - command ``/bin/grep '^\\s*CHANNELURL=' /etc/mail/spamassassin/channel.d/*.conf``
+=========================================================================================================
+"""
+import re
+from collections import OrderedDict
+
+from insights.core import CommandParser
+from insights.core.plugins import parser
+from insights.specs import Specs
+
+
+@parser(Specs.spamassassin_channels)
+class SpamassassinChannels(CommandParser):
+    """
+    Class for parsing the ``/bin/grep '^\\s*CHANNELURL=' /etc/mail/spamassassin/channel.d/*.conf``
+    command.
+
+    Attributes:
+        channels (OrderedDict): channels grouped by configuration file
+
+    Sample output of this command is::
+
+        /etc/mail/spamassassin/channel.d/sought.conf:CHANNELURL=sought.rules.yerp.org
+        /etc/mail/spamassassin/channel.d/spamassassin-official.conf:CHANNELURL=updates.spamassassin.org
+
+    Examples:
+        >>> type(spamassassin_channels)
+        <class 'insights.parsers.spamassassin_channels.SpamassassinChannels'>
+        >>> spamassassin_channels.channels
+        OrderedDict([('/etc/mail/spamassassin/channel.d/sought.conf', ['sought.rules.yerp.org']), ('/etc/mail/spamassassin/channel.d/spamassassin-official.conf', ['updates.spamassassin.org'])])
+    """
+    channels = None
+
+    def parse_content(self, content):
+        channels = OrderedDict()
+        for line in content:
+            file_name, file_line = line.split(":", 1)
+            channel = re.sub('^\\s*CHANNELURL=', '', file_line).strip()
+            if file_name not in channels:
+                channels[file_name] = []
+            channels[file_name].append(channel)
+
+        self.channels = channels
```

Comment:
What is the meaning of this line and line 33? I am just curious.

Reply:
If there is an exception, the `self.channels` will stay None. However using class variable is likely wrong, that should have been done in the init.